Include dashboard task creation in AI/BI dashboard DABs tutorial (#81)
databricks/bundle-examples: knowledge_base/dashboard_nyc_taxi/README.md

# Dashboard for NYC Taxi Trip Analysis

> This example shows how to define an AI/BI dashboard within a Databricks Asset Bundle, set up a job to capture a snapshot of the dashboard, and email it to a subscriber.

Reviewer: ```suggestion
This example shows how to define a Databricks Asset Bundle with an AI/BI dashboard and a job that captures a snapshot of the dashboard and emails it to a user.
```

Reviewer: ("demonstrates" is OK, but "shows" is fine too.)

> It deploys the sample __NYC Taxi Trip Analysis__ dashboard to a Databricks workspace and configures a daily schedule to run the dashboard and send the snapshot via email.

Reviewer: ```suggestion
It deploys the sample __NYC Taxi Trip Analysis__ dashboard to a Databricks workspace and configures a daily schedule to run the dashboard and send the snapshot in email to a specified email address.
```

For more information about AI/BI dashboards, please refer to the [documentation](https://docs.databricks.com/dashboards/index.html).

## Prerequisites

> - For creating dashboard: Databricks CLI v0.232.0 or above
> - For dashboard snapshot task: Databricks CLI v0.250.0 or above

Reviewer: ```suggestion
This example includes a dashboard snapshot task, which requires Databricks CLI  v0.250.0 or above. Creating dashboards in bundles is supported in Databricks CLI v0.232.0 or above.
```
Or, I would just list v0.250.0 because that is the prerequisites for this example.

## Usage

> #### Step1: Modify `databricks.yml`:
> 
> - Update the `host` field under `workspace` to the Databricks workspace to deploy to.
> - Update the `warehouse` field under `warehouse_id` to the name of the SQL warehouse to use.

Reviewer: ```suggestion
1. Modify `databricks.yml`:

    - Update the `host` field under `workspace` to the Databricks workspace to deploy to.
    - Update the `warehouse` field under `warehouse_id` to the name of the SQL warehouse to use.
```
I don't feel there is enough content to warrant new headings.

> #### Step2: Modify `resources/nyc_taxi_trip_analysis.job.yml`:
> 
> - Update the `user_name` field under `subscribers` to the dashboard subscriber's email

Reviewer: ```suggestion
2. Modify `resources/nyc_taxi_trip_analysis.job.yml`:

    - Update the `user_name` field under `subscribers` to the dashboard subscriber's email.
```

#### Step3: Deploy the dashboard

Run `databricks bundle deploy` to deploy the dashboard.

Run `databricks bundle open` to navigate to the deployed dashboard in your browser. Alternatively, run `databricks bundle summary` to display its URL.

> Note:
> At this moment, we have created the AI/BI dashboard and sets up the snapshot job to run daily at 8 AM, capture a snapshot of the dashboard, and send it via email to the specified subscriber.

Reviewer: ```suggestion

The AI/BI dashboard is created, and the snapshot job is set to run daily at 8 AM, which captures a snapshot of the dashboard, and sends it in email to the specified subscriber.
```

For more information about AI/BI dashboards, please refer to the [documentation](https://docs.databricks.com/dashboards/index.html).

## Prerequisites

- For creating dashboard: Databricks CLI v0.232.0 or above
- For dashboard snapshot task: Databricks CLI v0.250.0 or above

## Usage

#### Step1: Modify `databricks.yml`:

- Update the `host` field under `workspace` to the Databricks workspace to deploy to.
- Update the `warehouse` field under `warehouse_id` to the name of the SQL warehouse to use.

#### Step2: Modify `resources/nyc_taxi_trip_analysis.job.yml`:

- Update the `user_name` field under `subscribers` to the email of the subscriber of the dashboard

#### Step3: Deploy the dashboard

Run `databricks bundle deploy` to deploy the dashboard.

Run `databricks bundle open` to navigate to the deployed dashboard in your browser. Alternatively, run `databricks bundle summary` to display its URL.
Note:
At this moment, we have created the AI/BI dashboard, but the dashboard snapshot job is not fully ready yet since the dashboard id is only generated after the dashboard is created.

> #### Step 4: Update dashboard id in the dashboard snapshot job

Reviewer 2: This is not necessary. You can use `${resources.dashboards.nyc_taxi_trip_analysis.id}` and it will interpolate the correct value during deployment.

Author: updated, thanks!

> #### Step 4: Update dashboard id in the dashboard snapshot job

Reviewer 2: This is not necessary. You can use `${resources.dashboards.nyc_taxi_trip_analysis.id}` and it will interpolate the correct value during deployment.
